src/App.tsx

```diff
@@ -26,198 +152,70 @@ export const App: React.FC = () => {
-            </div>
-          </div>
-        </section>
-
-        <footer className="todoapp__footer" data-cy="Footer">
-          <span className="todo-count" data-cy="todosCounter">
-            4 items left
-          </span>
-
-          <nav className="filter" data-cy="Filter">
-            <a
-              data-cy="FilterLinkAll"
-              href="#/"
-              className="filter__link selected"
-            >
-              All
-            </a>
-
-            <a
-              data-cy="FilterLinkActive"
-              href="#/active"
-              className="filter__link"
-            >
-              Active
-            </a>
-            <a
-              data-cy="FilterLinkCompleted"
-              href="#/completed"
-              className="filter__link"
-            >
-              Completed
-            </a>
-          </nav>
+        {todosList.length > 0
+          && (
+            <TodosList
+              todos={filteredTodos}
+              setTodosList={setTodoList}
+              setDeleteError={setDeleteError}
+              deleteAll={deleteAll}
+            />
+          )}
+
+        {isAdding && (
+          <TodoAddCard
+            todo={{
+              id: 0,
+              title: newTodoTitle,
+              completed: false,
+              userId: user ? user.id : 0,
+            }}
+            isLoading={isAdding}
+          />
+        )}
+
+        {todosList.length > 0
+          && (
+            <Footer
+              leftTodos={leftTodos}
+              filterBy={filterBy}
+              setFilterBy={setFilterBy}
+              completedTodos={completedTodos}
+              deleteAllCompleted={deleteAllCompleted}
+            />
+          )}
+      </div>
 
+      {(titleError || deleteError || updateError || addingError) && (
+        <div
+          data-cy="ErrorNotification"
+          className="notification is-danger is-light has-text-weight-normal"
+        >
           <button
-            data-cy="ClearCompletedButton"
+            data-cy="HideErrorButton"
             type="button"
-            className="todoapp__clear-completed"
-          >
-            Clear completed
-          </button>
-        </footer>
-      </div>
+            className="delete"
+            onClick={ClearError}
+          />
 
-      <div
-        data-cy="ErrorNotification"
-        className="notification is-danger is-light has-text-weight-normal"
-      >
-        <button
-          data-cy="HideErrorButton"
-          type="button"
-          className="delete"
-        />
-
-        Unable to add a todo
-        <br />
-        Unable to delete a todo
-        <br />
-        Unable to update a todo
-      </div>
+          {titleError && ('Title can\'t be empty')}
+
+          {addingError && <br /> && ('Unable to add a todo')}
+
+          {deleteError && <br /> && 'Unable to delete a todo'}
+
+          {updateError && <br /> && 'Unable to update a todo'}
```

Comment:
I think better to create object with error data with error type and message

Then we'll be able to show message here instead of conditionally show this text here